### DISCUSSION TITLE
Erweiterung der benutzerdefinierten Werkzeuge für Gehzeitberechnungsservice

### DISCUSSION THREAD

**Discussion by ue-black** [e-netze/webgis-community#221] [Ideas]
Wir haben folgenden Wunsch ans Christkind:

Das VOIBOS-Gehzeitberechnungsservice soll in den Atlas eingebaut werden. 

Handbuch Service der VOIBOS Services:
[voibos_handbuch_user.pdf](https://github.com/user-attachments/files/17828215/voibos_handbuch_user.pdf)

Wir haben es wie folgt im custom.js konfiguriert: 

`tooltype: 'sketch1d'`

![image](https://github.com/user-attachments/assets/a2bed36c-5848-4c03-9bee-a3f451af691e)

Das heißt man muss selbst eine Linie einzeichnen. Das ist OK, wir haben aber in unserem Atlas sämtliche Wanderwege und natürlich das Straßennetz ebenfalls als gview Dienst drinnen. Das heißt die Geometrien sind da und ein Zusammensammeln der Geometrien wäre hier viel sinnvoller. Damit könnte man wirklich eigene länger Wanderungen nutzerfreundlich planen.
Eine bestehende Geometrie wird an das Voibos-Service geschickt.  D.h. viell. einen zusätzlichen tooltype definieren:

`tooltype: 'selection'`

und dann wenn man das Tool auswählt, erhält man einen Button mit "Auswahl übernehmen" und kann dann über ein Pop-Up auswählen ob man die bestehende Geometrie des Redlining oder einer Auswahl an den Voibos-Service übergibt.

Ungefähr so:
![image](https://github.com/user-attachments/assets/edb9faea-1beb-4326-ada9-8b7bfbf455f3)

Dazu bräuchte es dann wahrscheinlich zwei benutzerdefinierte Werkzeuge (eines für die Auswahl bestehender Objekte und eines wenn man selbst einen sketch zeichnen würde), das zu kombinieren wäre natürlich noch besser, aber wahrscheinlich Mehraufwand, oder?

Bzgl. Geometrie steht im Benutzhandbuch, dass Lücken zum Teil verbunden werden, das müsste man sich noch im  Detail ansehen welche Anforderungen die Geometrie haben muss die ans Service geschickt wird...

**Comment by gis-eni** [comment]
Das ist meiner Meinung schon recht kompliziert und was macht man wenn man mehrere Wege ausgewählt hat mit der Reihenfolge usw.
Warum nicht beim Zeichnen sagen: "Aus Geometrie" übernehmen?

**Reply by gis-eni** [reply]
Aus Anwendersicht würde ich mir eher wünschen, dass ich einen Wanderweg auswähle (mit Identify oder Suche) und von dort über einen Link die Anwendung aufrufen kann. 

**Hier könnte man sich als Erweiterung andenken, dass man in die Hotlinks auch die Geometrie in einer bestimmten Form übergeben kann. Allerdings muss man bedenken, dass die Links dann schnell einmal zu lang werden können!**

**Reply by ue-black** [reply]
Wie oben beschrieben geht es nicht um ein feature sondern eine Auswahl von mehreren features. Ein Wanderweg bei uns ist nur ein kurzer Abschnitt und keine Route vom Dorfzentrum zum Gipfelkreuz.

**Reply by gis-eni** [reply]
Genau und darum würde das auch mit der Selektion nicht funktionieren, weil wie schicke ich hier die Reihenfolge mit?

Und: Das würde ja auch mit der Link Variante funktionieren, weil es gibt ja auch die 1:n Links => Alle ``Ids`` aus der Ergebnistabelle werden übergeben.

**Reply by faulgurke** [reply]
Ausgezeichnet, das ist eine Variante die uns schon weiterhelfen würde. 
Wenn ich dich richtig verstehe wäre das folgender Ablauf:
1. Abfragen eines oder mehrerer Wegstücke.

![wanderwege_markiert](https://github.com/user-attachments/assets/6f34da68-77f3-449e-b0f1-f8cee61f1074)

2. Darstellung der Ergebnisse als Tabelle

![wanderwege_tabelle](https://github.com/user-attachments/assets/85edc884-e41c-481e-ac22-5aaee0a687be)

3. Ausführen eines noch einzubauenden Links, welcher die Geometrie eines oder alle Wegstücke an die Rechenraum-Anwendung übergibt.
Auf Seite 18 des voibos-Benutzungshandbuches (siehe oben) ist ein Beispiel-Aufruf aufgeführt.

4. Weil diese Variante einer Längenbeschränkung der Parameterübergabe unterliegt, wäre auch **folgende Variante denkbar**:
WebGIS produziert mit den ausgewählten Geometrien ein JSON File welches im Download Bereich des WebGIS zur Verfügung gestellt wird (eindeutiger filename). Anschließend kann man einen POST-Request an VOIBOS schicken, in welchem nur mehr der Namen des JSON files vorkommt (keine URL Längenbeschränkung).

**Reply by gis-eni** [reply]
Das mit Variante 4 eher nicht ;)

Aber grundsätzlich hat man da immer das Problem mit der Reihenfolge. Wenn man im WebGIS etwas abfragt ist die reihenfolge der Ergebnisse zufällig. Macht es bei diesem Tool dann überhaupt sind, das so zu machen?
